protocols/relay/src/behaviour.rs

```diff
@@ -0,0 +1,595 @@
+                            event,
+                        });
+                } else {
+                    let dest_id = req.dst_peer().peer_id;
+                    self.incoming_relay_reqs.entry(dest_id).or_default().push(
+                        IncomingRelayReq::DialingDst {
+                            request_id,
+                            req,
+                            src_id: event_source,
+                            src_addr,
+                        },
+                    );
+                    self.outbox_to_swarm
+                        .push_back(NetworkBehaviourAction::DialPeer {
+                            peer_id: dest_id,
+                            condition: DialPeerCondition::NotDialing,
+                        });
+                }
+            }
+            // Remote wants us to become a destination.
+            RelayHandlerEvent::IncomingDstReq(src, request_id) => {
+                let send_back = RelayHandlerIn::AcceptDstReq(src, request_id);
+                self.outbox_to_swarm
+                    .push_back(NetworkBehaviourAction::NotifyHandler {
+                        peer_id: event_source,
+                        handler: NotifyHandler::Any,
+                        event: send_back,
+                    });
+            }
+            RelayHandlerEvent::OutgoingRelayReqError(_dst_peer_id, request_id) => {
+                self.outgoing_relay_reqs
+                    .upgrading
+                    .remove(&request_id)
+                    .unwrap();
+            }
+            RelayHandlerEvent::OutgoingRelayReqSuccess(_dst, request_id, stream) => {
+                let send_back = self
+                    .outgoing_relay_reqs
+                    .upgrading
+                    .remove(&request_id)
+                    .map(|OutgoingUpgradingRelayReq { send_back, .. }| send_back)
+                    .unwrap();
+                send_back.send(Ok(stream)).unwrap();
+            }
+            RelayHandlerEvent::IncomingDstReqSuccess {
+                stream,
+                src_peer_id,
+                relay_addr,
+            } => {
+                self.outbox_to_transport
+                    .push(BehaviourToTransportMsg::IncomingRelayedConnection {
+                        stream,
+                        src_peer_id,
+                        relay_addr,
+                    })
+            }
+        }
+    }
+
+    fn poll(
+        &mut self,
+        cx: &mut Context<'_>,
+        poll_parameters: &mut impl PollParameters,
+    ) -> Poll<NetworkBehaviourAction<RelayHandlerIn, Self::OutEvent>> {
+        if !self.outbox_to_transport.is_empty() {
+            match self.to_transport.poll_ready(cx) {
+                Poll::Ready(Ok(())) => {
+                    self.to_transport
+                        .start_send(self.outbox_to_transport.pop().unwrap())
+                        .unwrap();
+                }
+                Poll::Ready(Err(_)) => unimplemented!(),
```

Comment:
This is an oversight, thanks for the catch.

`Poll::Ready(Err(mpsc::SendError))` can only happen when the `Relay` `NetworkBehaviour` is polled after the `RelayTransportWrapper` is dropped. `ExpandedSwarm` does not provide a guarantee on the drop order (e.g. `behaviour` before `network`). Though for this case I assume we can expect that `ExpandedSwarm` would not poll the `behaviour` after having dropped the `network`.

I will put a descriptive `panic` in place. Let me know in case my assumption above is incorrect.